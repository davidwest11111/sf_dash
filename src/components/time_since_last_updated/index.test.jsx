import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';
import moment from 'moment';
import 'moment-precise-range-plugin';

import TimeSinceLastUpdated from './index';


describe('outputs the expected tree when', () => {
  let wrapper;

  test('lastUpdated not passed', () => {
    wrapper = shallow((
      <TimeSinceLastUpdated />
    ));
  });

  test('lastUpdated is now', () => {
    wrapper = shallow((
      <TimeSinceLastUpdated lastUpdated={moment()} />
    ));
  });

  test('the update is not overdue', () => {
    wrapper = shallow((
      <TimeSinceLastUpdated lastUpdated={moment().subtract(1, 'seconds')} />
    ));
  });

  test('the update is overdue', () => {
    wrapper = shallow((
      <TimeSinceLastUpdated
        lastUpdated={moment().subtract(5, 'seconds')}
        msUntilWarning={3000}
      />
    ));
  });

  afterEach(() => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('instance methods', () => {
  let instance;
  beforeEach(() => {
    instance = shallow((
      <TimeSinceLastUpdated />
    )).instance();
  });

  test('sets up interval for forceUpdating time', () => {
    expect(instance.interval).toEqual(expect.any(Number));
  });
});
