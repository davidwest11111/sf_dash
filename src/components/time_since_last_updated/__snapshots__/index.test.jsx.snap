// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs the expected tree when lastUpdated is now 1`] = `
<div
  className="time-since-last-updated text-right"
>
  Just now.
</div>
`;

exports[`outputs the expected tree when lastUpdated not passed 1`] = `""`;

exports[`outputs the expected tree when the update is not overdue 1`] = `
<div
  className="time-since-last-updated text-right"
>
  1 second ago.
</div>
`;

exports[`outputs the expected tree when the update is overdue 1`] = `
<div
  className="time-since-last-updated text-right overdue"
>
  5 seconds ago.
</div>
`;
